video: fix dcmipp devicetree

ngphibang opened the pull request:
- Use normal child nodes for pipes: the pipe nodes are not video interfaces. Describe them as normal child nodes instead of using port/endpoint.
- Define video device for each pipe instead of only the main dcmipp device so that we can do set/get_ctrl()
- Another minor typo fix

ngphibang committed:
drivers: video: stm32_dcmipp: Fix a typo on driver instance name

The driver instance variable name is inst, not n.

Signed-off-by: Phi Bang Nguyen <phibang.nguyen@nxp.com>

avolmat-st commented on dts/arm/st/mp13/stm32mp135.dtsi:
```diff
@@ -21,25 +21,20 @@
 				#size-cells = <0>;
 
-				port@0 {
+				pipe_dump: pipe@0 {
 					reg = <0>;
-
-					endpoint {
-						remote-endpoint-label = "";
-						bus-type = <VIDEO_BUS_TYPE_PARALLEL>;
-					};
 				};
+			};
```
Yes, only the N6 has several pipe. Others have single pipe. Is that an issue to keep it as you propose with current patch ?
Aka always have pipes { } and below it one or more pipe@x entries ?

ngphibang replied:
Both ways works but I think it is more "correct" to remove the pipes {} node if we are sure that it has only one pipe and in the driver we treat it with `#if defined(STM32_DCMIPP_HAS_PIXEL_PIPES)` as alreadt done in some other places in the driver ?

avolmat-st replied:
> Both ways works but I think it is more "correct" to remove the pipes {} node if we are sure that it has only one pipe and in the driver we treat it with `#if defined(STM32_DCMIPP_HAS_PIXEL_PIPES)` as alreadt done in some other places in the driver ?

Yes ok for me. Would you update your PR to make this change as well ?

ngphibang committed:
drivers: video: stm32_dcmipp: Use normal child nodes for pipes

The pipe nodes are not video interfaces. Describe them as normal child
nodes instead of using port/endpoint.

Signed-off-by: Phi Bang Nguyen <phibang.nguyen@nxp.com>

ngphibang committed:
drivers: video: dcmipp: Define video device for each pipe

Define a video device for each pipe instead for the main dcmipp device
since the pipe is the device that communicates with application
(zephyr,camera chosen node), not the dcmipp device.

This helps to enable camera controls as well.

Signed-off-by: Phi Bang Nguyen <phibang.nguyen@nxp.com>

ngphibang commented:
`Could you maybe only modify minimum amount of lines and all other parts ?`

Yes, done.

I added a 3rd commit "drivers: video: dcmipp: Define video device for each pipe" as well.

Otherwise, I encountered a compilation error (due to the 2nd commit):

```
/home/nxf91161/zephyrproject/zephyr/drivers/video/video_stm32_dcmipp.c:1623:31: error: '#' is not followed by a macro parameter
 1623 | #define STM32_DCMIPP_INIT(inst)****
```

It took me quite a time to debug but still cannot figure out what does it mean this error (I didn't modify these lines at all ...).

Will come back to this later or would be glad if you can help :-).

avolmat-st commented on drivers/video/video_stm32_dcmipp.c:
```diff
@@ -1620,6 +1620,14 @@ static void stm32_dcmipp_isr(const struct device *dev)
+#define STM32_DCMIPP_PIPES(inst) \
+	DT_FOREACH_CHILD_VARGS(DT_INST_CHILD(inst, pipes), DCMIPP_PIPE_INIT_DEFINE, inst);
+#else
+#define STM32_DCMIPP_PIPE_INIT(node_id, inst) DCMIPP_PIPE_INIT_DEFINE(node_id, inst)
+#define STM32_DCMIPP_PIPES(inst) STM32_DCMIPP_PIPE_INIT(DT_INST_CHILD(inst, pipe), inst)
```
Thanks @etienne-lms for the review. While this sounds correct, this change won't compile, the preproc get confused in this case.

ngphibang replied:
yes, node_id need to be expanded before concatenation so we need one level of indirection here

avolmat-st reviewed (approved):
Thanks @ngphibang !!

etienne-lms reviewed (approved):
Looks consistent to me, AFAICT.
